gensim/models/nmf.py

```diff
@@ -580,22 +561,25 @@ def _setup(self, corpus):
         self.A = np.zeros((self.num_topics, self.num_topics))
         self.B = np.zeros((self.num_tokens, self.num_topics))
 
-        return corpus
-
     def update(self, corpus):
         """Train the model with new documents.
 
         Parameters
         ----------
-        corpus : iterable of list of (int, float) or `csc_matrix` with the shape (n_tokens, n_documents), optional
+        corpus : iterable of list of (int, float) or `csc_matrix` with the shape (n_tokens, n_documents)
             Training corpus.
             Can be either iterable of documents, which are lists of `(word_id, word_count)`,
             or a sparse csc matrix of BOWs for each document.
             If not specified, the model is left uninitialized (presumably, to be trained later with `self.train()`).
 
         """
-        if self._W is None:
-            corpus = self._setup(corpus)
+        if not (utils.is_corpus(corpus) or isinstance(corpus, scipy.sparse.csc.csc_matrix)):
+            raise ValueError(
+                "Corpus type should be either `gensim.corpus.IndexedCorpus` or `scipy.sparse.csc.csc_matrix`"
```

Comment:
That doesn't sound right -- we should accept any document stream, not just `IndexedCorpus`. (but see comment above -- I don't think we want this check at all)

Reply:
Agreed, removed the check.